lib/samples/show_legend/show_legend_sample.dart

```diff
@@ -0,0 +1,174 @@
+              // add a map view to the widget tree and set a controller.
+              child: ArcGISMapView(
+                controllerProvider: () => _mapViewController,
+                onMapViewReady: onMapViewReady,
+              ),
+            ),
+            // add a dropdown button to the widget tree.
+            DropdownButtonHideUnderline(
+              child: DropdownButton(
+                value: _selectedLegend,
+                menuMaxHeight: 200,
+                alignment: Alignment.center,
+                hint: const Text(
+                  'Legend',
+                  style: TextStyle(
+                    color: Colors.deepPurple,
+                  ),
+                ),
+                elevation: 16,
+                style: const TextStyle(color: Colors.deepPurple),
+                // no need to set up onChanged callback
+                onChanged: (n) {},
+                items: _legendsDropDown,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // get the screen scale
+    final screenScale = MediaQuery.of(context).devicePixelRatio;
+    // create an image layer
+    final imageLayer = ArcGISMapImageLayer.withUri(
+      Uri.parse(
+          'https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer'),
+    );
+    // create a feature table
+    final featureTable = ServiceFeatureTable.withUri(
+      Uri.parse(
+          'https://sampleserver6.arcgisonline.com/arcgis/rest/services/Recreation/FeatureServer/0'),
+    );
+    // create a feature layer
+    final featureLayer = FeatureLayer.withFeatureTable(featureTable);
+    // add image and feature layers to the opertaional layers list of the map
+    _map
+      ..operationalLayers.add(imageLayer)
+      ..operationalLayers.add(featureLayer);
+    // load the image and feature layers
+    await featureLayer.load();
+    await imageLayer.load();
+    // create a list to store operational layers and populate it with image and feature layers
+    final operationalLayersList = <LayerContent>[
+      ...imageLayer.subLayerContents,
+      featureLayer
+    ];
+
+    // get the legend info for each layer and add it to the legends dropdown list
+    for (var layer in operationalLayersList) {
+      // get the legend info for the current layer
+      final layerLegends = await layer.fetchLegendInfos();
+      // add the name of the current layer
+      _legendsDropDown.add(
+        DropdownMenuItem(
+          value: layerLegends.first,
+          child: Text(
+            layer.name,
+            style: const TextStyle(
+              fontSize: 15,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ),
+      );
+
+      // add current layer's legends to the dropdown list
+      for (var legend in layerLegends) {
+        try {
+          // create a swatch for the legend
+          _arcGISImage = await legend.symbol!.createSwatch(
```

Comment:
`_arcGISImage` shouldn't be a member variable. It should just be a temporary local variable declared right before the "try" (and not have an underscore).

Reply:
And instead of using `!` with `symbol!`, it should check if `symbol` is `null`. If it's `null`, it should handle having a null ArcGISImage down below. Instead of doing `_arcGISImage!.getEncodedBuffer()`, it should check if `_arcGISImage` is null and maybe substitute an Icon or something.